src/Engines/QueryBuilderEngine.php

```diff
@@ -394,7 +399,28 @@ private function getSearchKeyword($i, $raw = false)
 
         return $this->setupKeyword($keyword);
     }
-
+    
+    /**
+     * Check if model use SoftDeletes trait
+     *
+     * @return boolean
+     */
+    private function modelUseSoftDeletes()
+    {
+        $app = app();
+        
+        //check for laravel version
+        //in 4 and 5 versions SoftDeletes trait has a different names
+        //we can check only first number
+        if (strpos($app::VERSION, '5') === 0) {
+            $class = 'Illuminate\Database\Eloquent\SoftDeletes';
+        } else {
+            $class = 'Illuminate\Database\Eloquent\SoftDeletingTrait';
+        }
+        
+        return in_array($class, class_uses($this->query->getModel()));
```

Comment:
I not sure, that i understand your comment aright, but model is not a instance of SoftDeletes.
SoftDeletes - is only a trait, that uses in model and this check will be not correct.